config/initializers/tag2link.rb

```diff
@@ -0,0 +1,25 @@
+# A map of each OSM key to its formatter URL. For example:
+# { "ref:vatin" => "https://example.com/$1" }
+TAG2LINK = lambda {
+  # the JSON data is an array with duplicate entries, which is not efficient for lookups.
```

Comment:
Why do we need an immediately invoked lambda here? Just to avoid namespace pollution? You could actually write it all with only one variable other than the end result:

```ruby
TAG2LINK = JSON.parse(Rails.root.join("node_modules/tag2link/index.json").read)
               .reject { |item| item["rank"] == "deprecated" }
               .group_by { |item| item["key"] }
               .transform_keys { |key| key.sub(/^Key:/, "") }
               .transform_values { |items| items.max_by { |item| ranks.index(item["rank"]) }.fetch("url") }
```

which just needs the array of ranks...

Reply:
yeah I just didn't want to pollute the global namespace.

this is much more elegant, thanks. i'm not a ruby developer so any suggestions about code style are most welcome